test-runner/src/commands/safl/index.ts

```diff
@@ -0,0 +1,166 @@
+import type { CommandModule } from "yargs";
+
+import { performance, PerformanceObserver } from "perf_hooks";
+import { Worker } from "worker_threads";
+import { cpus } from "os";
+
+import { Init, ResponseMsg, WToPMsg } from "./types";
+
+interface Options {
+  connectionString: string;
+  client_count: number;
+  worker_count: number;
+  total_events_to_read: number;
+  report_per_number_of_events: number;
+}
+
+const safl: CommandModule<{}, Options> = {
+  command: "safl",
+  describe: "Multi Subscribe To All Processor",
+  builder: {
+    client_count: {
+      type: "number",
+      demandOption: true,
+    },
+    worker_count: {
+      type: "number",
+      default: cpus().length - 1,
+    },
+    total_events_to_read: {
+      type: "number",
+      default: 1000,
+    },
+    report_per_number_of_events: {
+      type: "number",
+      default: 1000,
+    },
+  },
+  handler,
+};
+
+async function handler({
+  client_count: clientCount,
+  worker_count,
+  total_events_to_read: totalEventsToRead,
+  report_per_number_of_events: reportPerNumberOfEvents,
+  connectionString,
+}: Options) {
+  const workerCount = Math.min(clientCount, worker_count);
+
+  const splitInteger = (num: number, parts: number) => {
```

Comment:
Note to self:

Should split all of this worker spawning into a utility, rather than copy pasting it everywhere